nidx/tests/test_shards_api.rs

```diff
@@ -105,6 +107,11 @@ async fn test_shards_create_and_delete(pool: sqlx::PgPool) -> anyhow::Result<()>
         sqlx::query!("UPDATE segments SET delete_at = NOW() WHERE index_id = $1", index_id as IndexId,)
             .execute(&meta.pool)
             .await?;
+
+        // Add a new segment as if it was coming from a merge job started before
+        // deletion and finished just before it. The segment has no deletion
+        // mark but shouldn't retain himself, its index or shard to be purged
+        Segment::create(&meta.pool, index_id, Seq::from(1000i64), 100, JsonValue::Null).await?;
```

Comment:
I'd mark the newly created segment as ready. Newly created segments are marked for deletion in the future so that might make this test less robust.

Reply:
Done